lib/Crypto/Crypt.php

```diff
@@ -451,9 +467,17 @@ protected function isValidPrivateKey($plainKey) {
 	 * @throws DecryptionFailedException
 	 */
 	public function symmetricDecryptFileContent($keyFileContents, $passPhrase, $cipher = self::DEFAULT_CIPHER, $version = 0, $position = 0) {
+		$this->eventDispatcher->addListener('files.aftersignaturemismatch', function (GenericEvent $event) {
```

Comment:
let's just use a static attribute... we already have this kind of dirty practice in the decrypt all command.

Reply:
hmm now thinking this is not possible because of the encryption module abstraction.

Reply:
idea would be if there was a way to inject generic flags or configs into the encryption module

in your first attempt you used the oc_appconfig table for that, which was a bit too global.

Reply:
The whole problem is here with the code flow 

transfer commnad <----> Encryption Wrapper <-----> Crypt 

The transfer needs to output the files which are problematic and then come back for a retry in the encryption wrapper to copy the file again and crypt should know that this time no signature check.

Even I was not fully satisfied with the event logic.

Will see what can be done here..